Feature branch/todo list

bini-i opened the pull request:
This is a todo-list project where a user can create a todo and organize them into projects

### The project has the following features:
* An option to create a new todo.
* An option to update a todo.
* An option to delete a todo.
* An option to create a project.
* An option to categorize todos into projects.

bini-i committed:
Add git workflows

- eslint and stylelint linters

MrSayyod committed:
Install webpack

MrSayyod committed:
Create initial class and properties

MrSayyod committed:
Create default array for todos when project is not selected

bini-i committed:
Add home.js

- renders form html and list of todolist using js

MrSayyod committed:
Add id for inputs to get data

bini-i committed:
Add factory.js and createTodo.js

- factory.js hosts todoModule and projectModule to encapsulate todo and
project state
- createTodo.js creates a todo using createTodo method from todoModule

bini-i committed:
update home.js

- add functionality to delete a todo with click eventlistner on todo
delete icon

bini-i committed:
Update factory.js

- add getProjectsTodos to get a todo which belongs to a project

bini-i committed:
Update home.js and index.js

- add option to create a project with todos
- display details of todos

MrSayyod committed:
Add edit/update modal form

MrSayyod committed:
Delete comments

MrSayyod committed:
Add Todo properties inot modal form

MrSayyod committed:
Update modal form

bini-i committed:
Update home.js and index.js

- add event listener to edit icon to update a todo

MrSayyod committed:
Fix stylelint and eslint errors

bini-i committed:
Update styling and add README

- used eslint for styling

voscarmv reviewed (changes requested):
# Impressive work. 💻 👀 ✨ 
- Good work getting the basic structure of the project right. Nice separation of DOM operations. Good code modulations.

## However, there are some changes required 👀 🧰 🔨 🔧 
- Please use localStorage to make the data persist between sessions on your project. Currently I lose all my data if I refresh the page.
- After I add different ToDos with the same title on different projects, if I try to delete a ToDo from one project, it sometimes deletes the ToDo in a different project. Please fix this.
- If I add some ToDos to the default project, and I delete the first ToDo repeatedly, sometimes the app will delete the last ToDo instead of the one I clicked on. Please fix this.
- Sometimes I can't delete the last ToDo from a project after adding multiple ToDos and trying to delete them all one by one.

**[OPTIONAL]**
- I'm able to add project with blank titles and descriptions. Please add some input validation to prevent this.
- Please add a LiveDemo link to your README.

### Submit these changes and request a new code review! 😄 👍 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

bini-i committed:
Fix factory.js (deleteTodo method)

- finds array index of todo to be deleted and splice(deletes) the todo

bini-i committed:
Refactor renderTodoList.js and Fix renderModal.js

- separate rendering of different sections into there own file under
renderDomElements folder
- Fix renderModal.js to check if a container DOM element exist before
creating one and remove it

bini-i committed:
Update index.js and renderForm.js

- added validation for form input fields using bootstrap validation

bini-i committed:
Update styling

- with eslint

bini-i committed:
Update README

- add live demo link

bini-i committed:
Update factor.js

- made todoList and project to be stored in localstorage

arslanbisharat reviewed (approved):
# Approved :clap:

Hello Developer, :wave:

Thank You for submitting this Pull Request. I am thrilled to announce that your project has met all the minimum requirements and now it is approved. :clap:

I am wishing you very good luck with the next project. :smile:

Once again, Congratulations!:joy:

**Best,
Arslan Bisharat(PT-TSE)
Slack: @arslanbisharat**

bini-i committed:
Update factory.js

- made deletion and update of todos be reflected on localstorage
